internal/pkg/csumio/csumwriter_test.go

```diff
@@ -97,3 +97,233 @@ func TestWriter_Success(t *testing.T) {
 		})
 	}
 }
+
+// TestWriter_WriteFactorCoverage exhaustively validates csumio.Writer.Write.
+// The test matrix uses pairwise coverage across three factors:
+//
+//	Factor A: requested write length p relative to chunkSize (> chunk / == chunk / < chunk).
+//	Factor B: remaining buffer capacity before Write relative to p (> p / == p / < p).
+//	Factor C: offset state (no chunk flushed yet vs. at least one chunk flushed).
+//
+// These three factors yield nine cases (case01–case09). Rows 1, 2, and 4 cannot occur in practice
+// because the buffer capacity never exceeds chunkSize, so those combinations are commented out
+// below. The remaining cases drive chunkSize±1 boundary writes to exercise every achievable
+// scenario.
+func TestWriter_WriteFactorCoverage(t *testing.T) {
+	chunkSize := csumio.MinimumChunkSize
+
+	pGreater := chunkSize + 1 // Represent p > chunkSize by overshooting by 1 byte.
+	pEqual := chunkSize       // Exact chunk boundary.
+	pSmaller := chunkSize - 1 // Represent p < chunkSize by undershooting by 1 byte.
+
+	almostFull := chunkSize - 1    // Leave only 1 byte of remaining capacity (capacity < p).
+	twoBytesShort := chunkSize - 2 // Expect chunk-2 tail during Close.
+	singleByte := 1                // Use single-byte writes to control remaining capacity precisely.
+
+	type matrixCase struct {
+		name              string
+		description       string
+		preLens           []int
+		targetLen         int
+		expectedChunkLens []int
+	}
+
+	cases := []matrixCase{
+		/* This case is omitted because "capacity > p" is impossible when capacity == chunkSize.
+		{
+			name:        "case01_pGreater_capacityChunk_offsetZero",
+			description: "p > chunk, capacity == chunk (approximates > p), offset zero",
+			preLens:     nil,
+			targetLen:   pGreater,
+			expectedChunkLens: []int{
+				chunkSize,
+				singleByte,
+			},
+		},
```

Comment:
I don't think we have to "approximate" anything.

```suggestion
		{
			name:        "case01_pGreater_capacityChunk_offsetZero",
			description: "p > chunk, capacity > p, offset zero",
		},
```